### DISCUSSION TITLE
Is there a way to save a coalesced gif?

### DISCUSSION THREAD

**Discussion by lucasvickers** [python-pillow/Pillow#7145] [Q&A] [answered]
Hi All,

I'm working on a project that requires fully coalesced gifs. Based on [this gist](https://gist.github.com/BigglesZX/4016539#file-gifextract-py-L28-L51), I'm able to confirm that I'm saving a non-coalesced gif.

Is there a current way to achieve this? If not, any ideas on what would need modification to allow this?

My code:
```python
  from PIL import Image as PilImage

  first = PilImage.open(image_files[0])
  sequence = []

  for i in range(1, len(image_files)):
    sequence.append(PilImage.open(image_files[i]))

  first.save('pillow-out.gif',
             save_all=True,
             append_images=sequence,
             optimize=False,
             compress_level=0)
```

And the currently resulting gif: 
![pillow-out](https://user-images.githubusercontent.com/1099950/236957384-8a122596-432b-41de-8a53-06fb09840da5.gif)

**Comment by radarhere** [comment]
The gist you linked to is 11 years old. We've significantly improved our reading of GIFs since then, so if you want to extract all of the frames from a GIF, you should no longer have to be concerned about pasting and palettes. You should just be able to
```python
from PIL import Image, ImageSequence
with Image.open("pillow-out.gif") as im:
  for i, frame in enumerate(ImageSequence.Iterator(im)):
    frame.save(str(i)+".png")
```

But your actual question is about saving. Sorry, I don't know exactly what you mean when you say 'coalesced'. Could you describe that in more detail for me? At the moment, I see 25 frames of the number 72 shrinking and then expanding again. What are you hoping to see?

Ideally, if you could attach your input images (`image_files`), that would also be helpful.

**Reply by lucasvickers** [reply]
When I say coalesced, I mean that every frame is an unoptimized complete set of pixels. This is different from frames that only contain the differential pixels. They describe it in [imagemagick docs](https://legacy.imagemagick.org/Usage/anim_basics/#coalesced), seems like it's not a universal term :) In the gist I referenced it's referred to as full vs additive, I want full.

So given these source images: [test.zip](https://github.com/python-pillow/Pillow/files/11432102/test.zip), I want a gif where each frame is a full 600x600 pixel image. And you are right, it's just a number pulsing against a gradient.

I'm only using the highlighted section of the gist ([L28-L51](https://gist.github.com/BigglesZX/4016539#file-gifextract-py-L28-L51)) that checks if the gif is full ( what I'm calling coalesced) or additive.

ImageMagick is able to do this using `-coalesce` but it's overall gif handling is poor and it renders some terrible artifacts: 
![wand-out](https://github.com/python-pillow/Pillow/assets/1099950/7dcb7585-4058-4f81-a91c-20bd8d562b80)

**Reply by radarhere** [reply, marked as answer]
Ah, ok. It's not possible to do this using Pillow's public API. However, you can workaround that by adding in the following code when generating your image.
```python
from PIL import GifImagePlugin
GifImagePlugin._getbbox = lambda a, b: (0, 0) + a.size
```

**Reply by lucasvickers** [reply]
Awesome, thank you!

**Comment by Yay295** [comment]
I think you're talking about the disposal method used. You can set this when saving. I think you probably want method 0.

https://pillow.readthedocs.io/en/stable/handbook/image-file-formats.html#gif-saving

**Reply by lucasvickers** [reply]
Thanks for the idea! I tried 0-3 but all of them have differential rendering (i.e. `update_region_dimensions != im.size`)
